Gems/Atom/Tools/MaterialCanvas/Code/CMakeLists.txt

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+if(NOT PAL_TRAIT_BUILD_HOST_TOOLS)
+    return()
+endif()
+
+o3de_pal_dir(pal_source_dir ${CMAKE_CURRENT_LIST_DIR}/Source/Platform/${PAL_PLATFORM_NAME} ${gem_restricted_path} ${gem_path} ${gem_parent_relative_path})
+
+include(${pal_source_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake) # PAL_TRAIT_ATOM_MATERIAL_CANVAS_APPLICATION_SUPPORTED
+
+if(NOT PAL_TRAIT_ATOM_MATERIAL_CANVAS_APPLICATION_SUPPORTED)
+    return()
+endif()
+
+ly_add_target(
+    NAME MaterialCanvas EXECUTABLE
+    NAMESPACE Gem
+    AUTOMOC
+    AUTORCC
+    FILES_CMAKE
+        materialcanvas_files.cmake
+        ${pal_source_dir}/platform_${PAL_PLATFORM_NAME_LOWERCASE}_files.cmake
+    PLATFORM_INCLUDE_FILES
+        ${pal_source_dir}/tool_dependencies_${PAL_PLATFORM_NAME_LOWERCASE}.cmake
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            .
+            Source
+            ${pal_source_dir}
+        PUBLIC
+            Include
+    BUILD_DEPENDENCIES
+        PRIVATE
+            Gem::AtomToolsFramework.Static
+            Gem::AtomToolsFramework.Editor
+            Gem::Atom_RHI.Public
+            Gem::Atom_RHI.Reflect
+            Gem::Atom_RPI.Edit
+            Gem::Atom_RPI.Public
+            Gem::Atom_Feature_Common.Public
+            Gem::Atom_Component_DebugCamera.Static
+            Gem::AtomLyIntegration_CommonFeatures.Static
+            Gem::GraphCanvasWidgets
+            Gem::GraphModel.Editor.Static
```

Comment:
It doesn't look like GraphModel is actually being used. Is that one of the follow-up steps?

Reply:
Correct. I have not done anything related to graph model yet. I have that as a follow up task once I create some test nodes and have a better understanding about how to do all of the interactions through graph model. 

TBD but if it's possible then I would like the graph canvas graph to be completely transient, maybe owned by the view, constructed from a minimal, material canvas specific document data as opposed to serializing all the entities and components that represent the graph. I don't know if this is possible but I'm envisioning the same thing for the nodes, that there would only be one concrete node class whose inputs, outputs, and other configuration can come from data, kind of like the module node. 

If we can make that abstraction work then it will be portable to other tools and datasets like the pass editor.